consensus_inputs.go

```diff
@@ -0,0 +1,258 @@
+package f3
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/filecoin-project/go-f3/certs"
+	"github.com/filecoin-project/go-f3/certstore"
+	"github.com/filecoin-project/go-f3/ec"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/filecoin-project/go-f3/internal/clock"
+	"github.com/filecoin-project/go-f3/manifest"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/ipfs/go-cid"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type gpbftInputs struct {
+	manifest  *manifest.Manifest
+	certStore *certstore.Store
+	ec        ec.Backend
+	verifier  gpbft.Verifier
+	clock     clock.Clock
+
+	ptCache *lru.Cache[string, cid.Cid]
+}
+
+func newInputs(manifest *manifest.Manifest, certStore *certstore.Store, ec ec.Backend,
+	verifier gpbft.Verifier, clk clock.Clock) gpbftInputs {
+	cache, err := lru.New[string, cid.Cid](256) // keep a bit more than 2x max ECChain size
+	if err != nil {
+		// panic as it only depends on the size
+		panic(fmt.Errorf("could not create cache: %w", err))
+	}
+
+	return gpbftInputs{
+		manifest:  manifest,
+		certStore: certStore,
+		ec:        ec,
+		verifier:  verifier,
+		clock:     clk,
+		ptCache:   cache,
+	}
+}
+
+func (h *gpbftInputs) getPowerTableCIDForTipset(ctx context.Context, tsk gpbft.TipSetKey) (cid.Cid, error) {
+	sTSK := string(tsk)
```

Comment:
How big do tipset keys get? do we want to hash these?

Reply:
on expectation ~5x the hash size